File: src/types/analytics.ts
export interface AnalyticsSummary {
  total_repos: number;
  public_repos: number;
  private_repos: number;
  active_repos: number;
  dormant_repos: number;
  dead_repos: number;
  empty_repos: number;
  archived_repos: number;
  total_stars: number;
  total_forks: number;
  total_size_kb: number;
}

export interface LanguageStat {
  language: string;
  count: number;
  percentage: number;
  color: string;
}

export interface GrowthPoint {
  month: string;
  count: number;
  cumulative: number;
}

export interface DecayPoint {
  month: string;
  active: number;
  dormant: number;
  dead: number;
}

export interface StarSnapshot {
  date: string;
  stars: number;
  repo_id: string;
}

export interface StarVelocityEntry {
  repo_name: string;
  stars_this_week: number;
  total_stars: number;
}

File: src-tauri/src/commands/analytics.rs
use crate::models::AppError;
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct AnalyticsSummary {
    pub total_repos: u32,
    pub public_repos: u32,
    pub private_repos: u32,
    pub active_repos: u32,
    pub dormant_repos: u32,
    pub dead_repos: u32,
    pub empty_repos: u32,
    pub archived_repos: u32,
    pub total_stars: u32,
    pub total_forks: u32,
    pub total_size_kb: u64,
}

#[derive(Serialize, Deserialize)]
pub struct LanguageStat {
    pub language: String,
    pub count: u32,
    pub percentage: f64,
}

#[tauri::command]
pub async fn analytics_get_summary() -> Result<AnalyticsSummary, AppError> {
    Ok(AnalyticsSummary {
        total_repos: 0,
        public_repos: 0,
        private_repos: 0,
        active_repos: 0,
        dormant_repos: 0,
        dead_repos: 0,
        empty_repos: 0,
        archived_repos: 0,
        total_stars: 0,
        total_forks: 0,
        total_size_kb: 0,
    })
}

#[tauri::command]
pub async fn analytics_get_language_distribution() -> Result<Vec<LanguageStat>, AppError> {
    Ok(vec![])
}
